test/test_connectionpool.py

```diff
@@ -247,6 +247,21 @@ def test_pool_close(self):
         with pytest.raises(Empty):
             old_pool_queue.get(block=False)
 
+    def test_pool_close_twice(self):
+        pool = connection_from_url('http://google.com:80')
+
+        # Populate with some connections
+        conn1 = pool._get_conn()
+        conn2 = pool._get_conn()
+        pool._put_conn(conn1)
+        pool._put_conn(conn2)
+
+        try:
+            pool.close()
+            pool.close()
```

Comment:
Presumably only the second call is expected to throw in the normal use case? In that case, let's remove the first call to `pool.close()` from the `try` block.

Can we also add some positive assertions? After the first call to `pool.close()` you can make some assertions about the `None`ness of the reference to the underlying pool.

Reply:
About both two points i wasnt sure.
1. Ok i will do
2. I let it out, becasuse its already tested inside the other test function and i want to prevent duplicated code, but if you want i will add this.

Reply:
It's just not great to have tests that never explicitly assert anything.